libs/filamat/src/eiff/BlobDictionary.h

```diff
@@ -39,27 +42,17 @@ class BlobDictionary {
         return mBlobs.size();
     }
 
-    // Returns the total storage size, assuming that each blob is prefixed with a 64-bit size.
-    size_t getSize() const noexcept {
-        return mStorageSize + 8 * getBlobCount();
-    }
-
     bool isEmpty() const noexcept {
         return mBlobs.size() == 0;
     }
 
-    const std::string& getBlob(size_t index) const noexcept {
-        assert(index < mBlobs.size());
-        return mBlobs[index];
+    std::string_view getBlob(size_t index) const noexcept {
+        return *mBlobs[index];
     }
 
 private:
-
-    // Use an ordered map with a transparent comparator to allow lookups using
-    // std::string_view without need for construction of a std::string object.
-    std::map<std::string, size_t, std::less<>> mBlobIndices;
-    std::vector<std::string> mBlobs;
-    size_t mStorageSize;
+    std::unordered_map<std::string_view, size_t> mBlobIndices;
+    std::vector<std::unique_ptr<std::string>> mBlobs;
```

Comment:
is the unique_ptr needed here? the std::string_view is not a reference to the std::string, it's a reference to the std::string's data.

I'm assuming a unique_ptr is used here so that the string_view stay valid when mBlobs is modified, if that's the case, I don't think a heap allocation is needed.

Reply:
The blobs vector grows over time, which causes strings to be copied (not moved) during a growth event, and the original string data is freed.  I actually wasn't sure those wrappers were needed either, but I saw use-after-delete errors without them.

I think it must be possible to write a vector implementation that doesn't do this, disappointed by this STL behavior.

Reply:
ah right -- I guess if it worked like I said, it would prevent the small string optimization. never mind then!